libparsec/crates/client/src/event_bus.rs

```diff
@@ -199,25 +199,131 @@ impl ToString for IncompatibleServerReason {
 // 1) Error-containing events are not often fired
 // 2) `anyhow::Error` itself is just a pointer, so converting it to an Arc is cheap
 impl_events!(
-    // Dummy event for tests only
-    Ping { ping: String },
-    // Events related to server connection
-    Offline,
+    // ***********************************************************************
+    // Server connection related events.
+    // ***********************************************************************
+
+    // Client-server communication is divided into two parts:
+    // - Client commands sent to the server through an RPC-like mechanism.
+    // - Server events sent to the client through a Server-Sent Events (SSE) mechanism.
+    //
+    // In practice we have a single SSE connection handled by the connection monitor, and
+    // multiple concurrent RPC commands depending on what is going on in the client (e.g.
+    // data synchronization, reading a file not in cache, etc.).
+    //
+    // On top of that, low-level details about the server connection are handled by the
+    // reqwest library (see `libparsec_client_connection` implementation), hence we have no
+    // guarantee on the number of physical TCP connections being used to handle RPC and SSE.
+    //
+    // However what we know is 1) RPC and SSE connect to the same server and 2) server
+    // and client both support HTTP/2. Hence we can expect all connections to be multiplexed
+    // over a single physical TCP connection.
+    //
+    // With this in mind, the choice has been to have the connection-related events only
+    // fired by the connection monitor: if a RPC command fails due to server disconnection,
+    // it is very likely the SSE connection will also fail right away.
```

Comment:
This could be moved in a module's doc comment or in the docs folder, otherwise we may never read it again ^^

Reply:
Totally, I'm thinking of adding this directly in `libparsec/README.md` so that we describe here the architecture of `libparsec` (the goal of the main crate, the layout of the sub-crates and the high-level view of how `libparsec_client` works)